File: packages/react-native/android/src/main/java/admob/plus/rn/ads/GenericAd.java
package admob.plus.rn.ads;

import admob.plus.rn.ExecuteContext;

public interface GenericAd {
    default boolean isLoaded() {
        return false;
    };

    default void load(ExecuteContext ctx) {
        ctx.error("Not implemented");
    };

    default void show(ExecuteContext ctx) {
        ctx.error("Not implemented");
    };

    default void hide(ExecuteContext ctx) {
        ctx.error("Not implemented");
    }
}

File: packages/capacitor/android/src/main/kotlin/admob/plus/core/GenericAd.kt
package admob.plus.core

interface GenericAd {
    val isLoaded: Boolean
        get() {
            Helper.Companion.NOT_IMPLEMENTED()
            return false
        }

    fun load(ctx: Context?) {
        Helper.Companion.NOT_IMPLEMENTED()
    }

    fun show(ctx: Context?) {
        Helper.Companion.NOT_IMPLEMENTED()
    }

    fun hide(ctx: Context?) {
        Helper.Companion.NOT_IMPLEMENTED()
    }
}
